src/plots/plots.js

```diff
@@ -809,6 +809,16 @@ plots.purge = function(gd) {
     // remove modebar
     if(fullLayout._modeBar) fullLayout._modeBar.destroy();
 
+    if(fullLayout._plots) {
```

Comment:
Thanks @etpinard!  This 
```
Plotly.newPlot = function(gd, data, layout, config) {
    gd = getGraphDiv(gd);

    // remove gl contexts
    Plots.cleanPlot([], {}, gd._fullData || {}, gd._fullLayout || {});
    
    Plots.purge(gd);
    return Plotly.plot(gd, data, layout, config);
};
```
seems to go ahead and ultimately call `scene2d` `destroy` method which has my few lines of trace line disposals. I'll double-check via remeasuring the memory footprint, but for that test I have to cherrypick it into the local branch which also has the undo history limit.